TeachingRecordSystem/src/TeachingRecordSystem.Dqt/DataverseAdapter.UpdateTeacher.cs

```diff
@@ -273,19 +273,27 @@ public UpdateTeacherHelper(DataverseAdapter dataverseAdapter, UpdateTeacherComma
                 Guid ittProviderId,
                 string slugId)
         {
-            // if SlugId is not passed in
-            // Find an active ITT record for the specified ITT Provider.
-            // otherwise find active ITT for the slugid (which should be one).
+            // if SlugId is passed in, use slugid for matching
+            // otherwise fallback to matching on establishmet
             // The record should be at the InTraining or Deferred or Withdrawn status unless the programme is 'assessment only',
             // in which case the status should be UnderAssessment or Deferred or withdrawn.
 
             List<dfeta_initialteachertraining> matching = new List<dfeta_initialteachertraining>();
-            var activeForProvider = string.IsNullOrEmpty(slugId) ? ittRecords
-                .Where(r => r.dfeta_EstablishmentId?.Id == ittProviderId && r.StateCode == dfeta_initialteachertrainingState.Active)
-                .ToArray()
-                : ittRecords
-                .Where(r => r.dfeta_SlugId == slugId && r.StateCode == dfeta_initialteachertrainingState.Active)
-                .ToArray();
+            var activeForProvider = Array.Empty<dfeta_initialteachertraining>();
+
+            if (!string.IsNullOrEmpty(slugId))
+            {
+                activeForProvider = ittRecords
+                    .Where(r => r.dfeta_SlugId == slugId && r.StateCode == dfeta_initialteachertrainingState.Active)
+                    .ToArray();
+            }
+
+            if (activeForProvider.Length == 0)
+            {
+                activeForProvider = ittRecords
+                    .Where(r => r.dfeta_EstablishmentId?.Id == ittProviderId && r.StateCode == dfeta_initialteachertrainingState.Active)
```

Comment:
I think we want a `&& string.IsNullOrEmpty(r.dfeta_SlugId)` too (to ensure we don't match on a different slug ID)